### PR TITLE
[vpj] Enable VenicePushJob to read from any File System

### PR DESCRIPTION
**Pull request by modithah**
## [vpj]Enable VenicePushJob to read from any File System
This PR enables the VenicePushJob to read data from any File System using the Hadoop file system abstraction.

- Bump hadoop to `2.10.2`
- Replace all FileSystem creations to get from the desired path instead of the default one
- The relevant fileSystem-specific jars should be available in the classpath
- Tested with the sample job and following hadoop-overrides and `hadoop-aws:2.10.2` and `aws-java-sdk-bundle-1.11.271.jar` provided in the classpath. Default AWS authentication chain was used. Credentials were available as environment variables.

> ```
> input.path=s3a://<PATH_TO_DATA>
> ```

- Theoretically this should enable reading data from any file system supported by hadoop

Resolves https://github.com/linkedin/venice/issues/705

## How was this PR tested?
- Existing unit and integration tests
- Tested with S3

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/utils/HadoopUtils.java [hunk 1]
```diff
@@ -52,6 +54,22 @@ public static void cleanUpHDFSPath(String path, boolean recursive) {
...
 
+  /**
+   * Set Hadoop configuration from Venice properties.
+   * @param conf Configuration to be updated
+   * @param props Venice properties
+   */
+  public static void setHadoopConfigurationFromProperties(Configuration conf, VeniceProperties props) {
+    for (String key: props.keySet()) {
+      String lowerCase = key.toLowerCase();
+      if (lowerCase.startsWith(HADOOP_PREFIX)) {
```
Comment (nisargthakkar) [inline clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/utils/HadoopUtils.java:65]: I'm not sure if we should use `hadoop-conf.` prefix for this as it is the prefix used for the main MR job. There is another MR job in VPJ that validates schemas of all files and builds the dictionaries. That MR job will not be able to read from S3.

Can we create a new one (maybe `hadoop-overrides.` or something better) and in the code either:
1. Extract these general overrides and copy them to `hadoop-conf.` and `hadoop-dict-build-conf.`; or
2. Where we extract configs with these two existing prefixes, we also parse extract from the general overrides prefix

Comment (modithah) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/utils/HadoopUtils.java]: Removed the usage. These configuration overrides are no longer needed. They should be picked up from the hadoop environment.

**Comment by modithah** [comment]
We should always be getting the FileSystem from the path in all new usages. Otherwise those places won't be able to use other FileSystems. I replaced all such existing usages

**Review by FelixGV — APPROVED** [review]
Thanks a lot! LGTM! Let's wait until CI finishes and I think it'll be ready to merge...

**Pull request merged**